### PR TITLE
Issue236 use database url

### PR DESCRIPTION
**Pull request by raar1**
Fixes https://github.com/PerfectFit-project/virtual-coach-issues/issues/236

### CODE REVIEW

File: helper/helper.py [hunk 1]
```diff
@@ -3,11 +3,27 @@
...
 
-def get_db_session(db_url=DB_URL):
-    engine = create_engine(db_url)
+    We are doing this string replacement because the config var DATABASE_URL provided by
+    Heroku may change.
+    """
+    if db_url.startswith("postgres://"):
+        db_url = db_url.replace("postgres://", "postgresql://", 1)
+
+    return db_url
```
Comment (svenvanderburg) [inline helper/helper.py:21]: Could you make use of this function in `dbschema/migrations/env.py`?

File: helper/populate_db.py [hunk 2]
```diff
@@ -2,16 +2,9 @@
...
 
-from dotenv import load_dotenv
-
 from dbschema.models import Users, ClosedUserAnswers, UserInterventionState, DialogQuestions, DialogAnswers
 from helper import get_db_session
 
-# load .env-file and get db_host and niceday_api_endopint variables
-load_dotenv()
-DB_HOST = os.getenv('DB_HOST', default='db:5432')
-
```
Comment (svenvanderburg) [inline helper/populate_db.py:13]: Why did you remove loading `dotenv` file? I think it would be nice to allow developers to easily set the db host url.

File: README.md [hunk 3]
```diff
@@ -55,7 +55,7 @@ Currently this is handled by running a script in the [onboarding/ dir of the vir
 Important: The database docker-compose must be up and running with all migrations applied, as described above, 
 BEFORE running any of the following steps.
 
-In addition, create a `.env` file in the root of this repo with an environment variable called `DB_HOST`
+In addition, create a `.env` file in the root of this repo with an environment variable called `DATABASE_URL`
 that refers to the endpoint of the database (usually `localhost:5432` if you're running things locally).
```
Comment (svenvanderburg) [inline README.md:59]: Probably this information now needs to be updated right? Because you now need to pass the full url? Please update documentation with the sensible defaults for developers.

Comment (svenvanderburg) [reply README.md:59]: I'm here referring to 'usually `localhost:5432` if you're running things locally'.

**Review by svenvanderburg — APPROVED** [review]
Nice! It seems like this won't break `virtual-coach-main` if we merge it together with https://github.com/PerfectFit-project/virtual-coach-main/pull/28#pullrequestreview-932993553 right?

File: helper/helper.py (hunk 1 above)
Comment (raar1) [reply helper/helper.py:21]: We were indeed considering that, but the problem is that this gives an 'out of package' import which isn't really safe/clean either, so for now it's probably best to just duplicate it (these will be the only 2 instances) with appropriate comment.

File: helper/populate_db.py (hunk 2 above)
Comment (raar1) [reply helper/populate_db.py:13]: It's because there is no .env on heroku, so everything is read via environment variables. Locally, we either use the docker-compose.yml (which loads the .env file for the container already, meaning anything set in .env is available to `os.getenv()`), or people might launch an individual container with docker, in which case they should use the `--env-file` arg to do the same.

File: README.md (hunk 3 above)
Comment (raar1) [reply README.md]: Agreed, I've updated the URL and made clear in the text that it assumes you're running it with (one of) the docker-compose yamls.

File: helper/populate_db.py (hunk 2 above)
Comment (svenvanderburg) [reply helper/populate_db.py:13]: Well, you might also just want to run things in your native environment. Then reading in a .env file is necessary (or at least very handy). The nice thing about dotenv is that if there is not .env file, but instead you have environment variables (like in a heroku deploy) it still works :)

File: helper/helper.py (hunk 1 above)
Comment (svenvanderburg) [reply helper/helper.py:21]: Check

File: helper/populate_db.py (hunk 2 above)
Comment (raar1) [reply helper/populate_db.py:13]: That's true, but we haven't really supported running it outside of docker for a while now. My worry with this is that it might cause confusion when running with the docker-compose (since it silently reads in a .env whether you've configured docker-compose to use it or not, and might overwrite other vars). But I don't feel so strongly about it that I will refuse to put it back in :)

Comment (raar1) [reply helper/populate_db.py:13]: (Hmm and there is a `python-dotenv` package that runs scripts locally and sets the .env as environment that could also maybe used during local dev, so that's maybe a possibility for people who still want to run outside of docker)
